File: server/src/main/java/com/smartexpense/api/application/dto/response/AuthResponseDto.java
package com.smartexpense.api.application.dto.response;

import lombok.Builder;
import lombok.Data;

import java.math.BigDecimal;
import java.util.UUID;

@Data
@Builder
public class AuthResponseDto {
    private String token;
    private UserDto user;

    @Data
    @Builder
    public static class UserDto {
        private UUID id;
        private String email;
        private String displayName;
        private String avatarUrl;
        private BigDecimal monthlyBudget;
        private Integer streakDays;
        private String themeColor;
    }
}

File: android/app/src/main/kotlin/com/smartexpense/android/data/remote/dto/response/AuthResponseDto.kt
package com.smartexpense.android.data.remote.dto.response

import com.google.gson.annotations.SerializedName

data class AuthResponseDto(
    @SerializedName("token") val token: String,
    @SerializedName("refreshToken") val refreshToken: String,
    @SerializedName("userId") val userId: String,
    @SerializedName("displayName") val displayName: String,
    @SerializedName("email") val email: String
)
